sql/src/main/java/io/crate/execution/engine/collect/files/FileReadingIterator.java

```diff
@@ -201,18 +201,27 @@ private void advanceToNextFileInput() throws IOException {
         Predicate<URI> uriPredicate = generateUriPredicate(fileInput, fileUri.globPredicate);
         List<URI> uris = getUris(fileInput, fileUri.uri, fileUri.preGlobUri, uriPredicate);
 
-        if (uris.size() > 0) {
+        if (uris.isEmpty()) {
+            LOGGER.warn("No files matched the pattern '{}'", fileUri.uri);
```

Comment:
it's a common case that a file is only on a single node - I don't think that we should log a warning on all others.

Reply:
If you use `/dir/*` then you won't have any missing files. If you use a hardcoded URL then I think you're doing something wrong. For example, you might have forgotten to copy the file to the node.

Reply:
Have to correct myself, of course it also logs for `/dir/*` here. That is questionable.